mailbox/store/src/main/java/org/apache/james/mailbox/store/search/MessageSearches.java

```diff
@@ -421,37 +421,54 @@ private boolean matches(SearchQuery.HeaderCriterion criterion, MailboxMessage me
      */
     private boolean matchesAddress(SearchQuery.AddressOperator operator, String headerName,
                                    MailboxMessage message) throws MailboxException, IOException {
-        String text = operator.getAddress().toUpperCase(Locale.US);
+        String text = operator.getAddress();
         List<Header> headers = ResultUtils.createHeaders(message);
         for (Header header : headers) {
             String name = header.getName();
             if (headerName.equalsIgnoreCase(name)) {
                 String value = header.getValue();
-                AddressList aList = LenientAddressParser.DEFAULT.parseAddressList(value);
-                for (Address address : aList) {
-                    if (address instanceof Mailbox) {
-                        if (AddressFormatter.DEFAULT.encode((Mailbox) address).toUpperCase(Locale.US)
-                            .contains(text)) {
-                            return true;
-                        }
-                    } else if (address instanceof Group) {
-                        MailboxList mList = ((Group) address).getMailboxes();
-                        for (Mailbox mailbox : mList) {
-                            if (AddressFormatter.DEFAULT.encode(mailbox).toUpperCase(Locale.US)
-                                .contains(text)) {
-                                return true;
-                            }
-                        }
-                    }
+                AddressList addressList = LenientAddressParser.DEFAULT.parseAddressList(value);
+                if (matchesAddress(addressList, text)) {
+                    return true;
                 }
 
                 // Also try to match against raw header now
-                return value.toUpperCase(Locale.US).contains(text);
+                return value.toUpperCase(Locale.US).contains(text.toUpperCase(Locale.US));
```

Comment:
maybe have a `upCase` method which does the `string.toUpperCase(Local.US)`

Reply:
not sure about the value it brings.